design/src/rfcs/rfc0012_unsynchronized_versioning.md

```diff
@@ -0,0 +1,207 @@
+service version number (this requirement has already been implemented).
+
+### Yanking
+
+It must be possible to yank an entire release with a single action. The publisher tool must
+be updated to understand which crate versions were released with a given release tag, and be able to
+yank all the crates published from that tag.
+
+Phase 1: Unsynchronized Dev Preview
+-----------------------------------
+
+Phase 1 will address the following challenges introduced by uncoordinating the `major.minor` versions:
+- Tracking of versions associated with a release tag
+- Creation of version bump process for code generated crates
+- Enforcement of version bump process in runtime crates
+- Yanking of versions associated with a release tag
+
+### Version Tracking
+
+A new manifest file will be introduced in the root of [aws-sdk-rust] named `versions.toml` that describes
+all versioning information for any given commit in the repository. In the main branch, the `versions.toml`
+in tagged commits will become the source of truth for which crate versions belong to that release, as well
+as additional metadata that's required for maintaining version process in the future.
+
+The special `0.0.0-smithy-rs-head` version that is used prior to Phase 1 for maintaining the runtime crate
+versions will no longer be used (as detailed in [Versioning for Runtime Crates](#versioning-for-runtime-crates)).
+
+This format will look as follows:
+```toml
+smithy_rs_version = "<release-tag|commit-hash>"
+
+[aws-smithy-types]
+version = "0.50.1"
+
+[aws-config]
+version = "0.40.0"
+
+[aws-sdk-s3]
+version = "0.89.0"
+model_hash = "<hash>"
+
+# ...
+```
+
+The auto-sync tool is responsible for maintaining this file. When it generates a new SDK, it will take
+the version numbers from runtime crates directly, and it will use the rules from the next section to determine
+the version numbers for the generated crates.
+
+### Versioning for Code Generated (SDK Service) Crates
+
+Code generated crates will have their `minor` version bumped when the version of [smithy-rs] used to generate
+them changes, or when model updates with API changes are made. Three pieces of information are required to
+handle this process: the previously released version number, the [smithy-rs] version used to generate the code,
+and the level of model updates being applied. For this last one, if there are multiple model updates that
+affect only documentation, but then one model update that affects an API, then as a whole they will be
+considered as affecting an API and require a `minor` version bump.
+
+The previously released version number will be retrieved from crates.io using its API. The [smithy-rs] version
+used during code generation will become a build artifact that is saved to `versions.toml` in [aws-sdk-rust].
+During phase 1, the tooling required to know if a model is a documentation-only change will not be available,
+so all model changes will result in a `minor` version bump during this phase.
```

Comment:
Are we ever going to run into a situation where we start releasing a version and then it chokes on the crates.io upload because of some problem with our code? If yes, would the versioning get messed up?